audb/core/publish.py

```diff
@@ -87,11 +102,56 @@ def job(file):
         progress_bar=verbose,
         task_description='Find media',
     )
-    deps._add_or_update_media(db_root, update_media)
+    if update_media:
+        deps._update_media(update_media)
+    if add_media:
+        deps._add_media(add_media)
```

Comment:
This does not directly solve https://github.com/audeering/audb/issues/143 as updating the dataframes are still happening outside of the "Find media" loop. Which makes sense as the goal was to speed this up by setting it once and not inside a loop. I also don't see a direct solution for it besides not using a dataframe.

So, maybe we show here a text statement like we did it in earlier versions of `audb` instead? The problem is, that users still might think it has crashed when it takes 60 minutes for this step.

Reply:
> This does not directly solve #143 as updating the dataframes is still happening outside of the "Find media" loop.

But are you sure that updating the dataframe was the bottleneck or calculating the file duration? Cause the latter is now done within fine media.

Reply:
Maybe you can run your test again?

Reply:
Ah,  yes, you are right, the duration was not calculated in the loop before.
I will test again.

Reply:
I did another check. You are totally right:

* "Find media": 110 minutes
* Assigning the dataframes: 10 seconds
* "Put media": 54 seconds (to local file system)

The only thing that could be improved from a user perspective on the progress bar is that it can have different progress rates for the old data and the new one added. E.g. in this case it takes <1 minute to progress to 80%, and then 110 minutes for the last 20 minutes. But it's not a big issue as you still will see progress with every file, so it's very unlikely that a user thinks it has crashed.

So I would propose that we just go with your changes here.

Reply:
You can also add "Closes #143" in the description.

Reply:
> You can also add "Closes #143" in the description.

done

Reply:
> So I would propose that we just go with your changes here.

ok